subsys/emds/emds_flash_new.c

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/drivers/flash.h>
+#include "emds_flash_new.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(emds_flash, CONFIG_EMDS_LOG_LEVEL);
+
+static int format_partitions_ee(struct emds_desc *desc)
+{
+	int rc;
+	struct flash_pages_info info;
+	int available_pages;
+
+	rc = flash_get_page_info_by_offs(desc->fa->fa_dev, desc->fa->fa_off, &info);
+	if (rc) {
+		LOG_ERR("Unable to get page info");
+		return -EINVAL;
+	}
+
+	available_pages = (desc->fa->fa_size + desc->fa->fa_off - info.start_offset) / info.size;
+	if (available_pages < 2) {
+		LOG_ERR("Not enough flash pages available for EMDS storage");
+		return -ENOSPC;
+	}
+
+	desc->part[0].part_off = info.start_offset;
+	desc->part[0].part_size = available_pages / 2 * info.size;
+	desc->part[1].part_off = desc->part[0].part_off + desc->part[0].part_size;
+	desc->part[1].part_size = desc->part[0].part_size;
+
```

Comment:
May be add a "TODO" here to indicate that actual earse API is not yet called here?

Reply:
Because of word "format". Nevermind.